lib/marketingcloudsdk/client.rb

```diff
@@ -105,6 +105,9 @@ def initialize(params={}, debug=false)
 				self.base_api_url = !(client_config["base_api_url"].to_s.strip.empty?) ? client_config["base_api_url"] : 'https://www.exacttargetapis.com'
 				self.request_token_url = !(client_config["request_token_url"].to_s.strip.empty?) ? client_config["request_token_url"] : 'https://auth.exacttargetapis.com/v1/requestToken'
```

Comment:
if useOauth2 is true, we need to add a validation to see if there is Auth endpoint specified in config or params. If not, we need to throw exception

Reply:
Added validation.